conformance/tests/httproute-matching-combinations.yaml

```diff
@@ -0,0 +1,84 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: matching-combinations
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
```

Comment:
Thank you @gauravkghildiyal! WDYT about adding another rule with all the possible `matches` set (path, header, method, queryparam)? If I see it correctly, we have many combinations, but none of them specifies all the possible `match` types.

Reply:
Sure thing! Done. Thanks for the review